docs/docs/08-guides/03-react-vite-websockets.md

```diff
@@ -0,0 +1,294 @@
+---
+title: React, Vite & Websockets 
+id: react-vite-websockets
+keywords: [Websockets, React, Vite, Local, Wing]
+---
+
+# React, Vite & Websockets 
+
+In this guide, we will go through the process of building a simple React webpage that connects to a Wing backend.
+
+The webpage will showcase a counter that can be incremented by clicking on it. This counter will increment a distributed counter deployed via the Wing backend. We will also demonstrate how to get real-time updates from the backend to the browser every time the counter is updated from any other webpage, using Websockets.
+
+## Prerequisites
+
+- Node.js v20 or later.
+- Wing has extended support for two IDEs. They provide syntax highlighting, completions, go-to-definition, etc. and embedded Wing Console support:
+
+  - [VSCode](https://marketplace.visualstudio.com/items?itemName=Monada.vscode-wing) - Official extension
+  - [IntelliJ](https://plugins.jetbrains.com/plugin/22353-wing) - Community extension
+
+## Step 1 - Installation & Scaffolding
+In this step, we will be creating our project by setting up two folders:
+
+ - `backend` - The place for Wing's backend code
+ - `frontend` - The place for our React app code
+
+### Creating a React App with Vite
+
+1. Create the root project folder:
+ ```sh
+ mkdir /tmp/confjs-wing-react
+ cd /tmp/confjs-wing-react
+ ```
+2. Create a new React app using Vite under the frontend folder:
+ ```sh
+ npm create vite frontend -- --template react-ts
+ ```
+3. Run your React app and ensure it works:
+ ```sh
+ cd frontend
+ npm install
+ npm run dev
+ ```
+> The result should be a very simple webpage that runs locally and works without any backend (note the counter is not shared between tabs).
```

Comment:
is there a counter at this point?

Reply:
yes, it is the default react app that vite provides